core/modules/url/store/actions.ts

```diff
@@ -0,0 +1,100 @@
+import { UrlState } from '../types/UrlState'
+import { ActionTree } from 'vuex';
+import * as types from './mutation-types'
+// you can use this storage if you want to enable offline capabilities
+import { cacheStorage } from '../'
+import queryString from 'query-string'
+import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
+import { processDynamicRoute, normalizeUrlPath } from '../helpers'
+import { storeCodeFromRoute, removeStoreCodeFromRoute } from '@vue-storefront/core/lib/multistore'
+
+const _parametrizedRoutedata = (routeData, query, storeCodeInPath) => {
+  routeData.params = Object.assign({}, routeData.params || {}, query)
+  if (storeCodeInPath && !routeData.name.startsWith(storeCodeInPath + '-')) {
+    routeData.name = storeCodeInPath + '-' + routeData.name
+  }
+  return routeData
+}
+// it's a good practice for all actions to return Promises with effect of their execution
+export const actions: ActionTree<UrlState, any> = {
+  // if you want to use cache in your module you can load cached data like this
+  registerMapping ({ commit }, { url, routeData }: { url: string, routeData: any}) {
+    return new Promise ((resolve, reject) => {
+      commit(types.REGISTER_MAPPING, { url, routeData })
+      cacheStorage.setItem(url, routeData).then(result => {
+        resolve(routeData)
+      }).catch(() => reject())
+    })
+  },
+  /**
+   * Register dynamic vue-router routes
+   */
+  registerDynamicRoutes ({ state, dispatch }) {
+    if (state.dispatcherMap) {
+      for (const [url, routeData] of Object.entries(state.dispatcherMap)) {
+        processDynamicRoute (routeData, url)
+        dispatch('registerMapping', { url, routeData })
+      }
+    }
+  },
+  mapUrl ({ state, dispatch }, { url, query }: { url: string, query: string}) {
+    const parsedQuery = typeof query === 'string' ? queryString.parse(query) : query
+    const storeCodeInPath = storeCodeFromRoute(url)
+    url = normalizeUrlPath(url, true)
+
+    return new Promise ((resolve, reject) => {
+      if (state.dispatcherMap[url]) {
+        return resolve (_parametrizedRoutedata(state.dispatcherMap[url], query, storeCodeInPath))
```

Comment:
_parametrizedRoutedata modifies state in this case. This should be a separate action, which modifies state by mutation

Reply:
It shouldn't - fixed with `Object.assign({}, routeData)`